### DISCUSSION TITLE
How can I use gtk2/3 without msys2?

### DISCUSSION THREAD

**Discussion by TexDh** [ruby-gnome/ruby-gnome#1471] [Q&A]
Hi, 
There used to be gtk2/3 and all the supporting gems with the corresponding binaries. But now  msys2 seems to be 'the' way to go.
When size is a constraint msys2 eats up most of the storage.
I'm in need for the gems those contain the binaries for Windows. So, I can avoid msys2.

I know the move to msys2 has been discussed before.

Is there a way to use gtk2/3 without installing msys2?

Thanks in advance.

Best regards
Tex

**Comment by kojix2** [comment]
Hello.
You mention that msys2 eats up most of the storage. 
Does this mean that the computer's internal hard drive will be full? For example, are you using a very small computer? Or is it a problem with portable strage devices? For example, you have the Ruby runtime environment on a USB stick?

**Reply by TexDh** [reply]
Hi Kojix,
I have a standalone runtime environment with a dedicated storage space (xxGB) in an internal drive. On each run, it generates quite some data (xxMB). 
In my case msys2 is about 1.5GB. 
With some iterations this storage becomes full.

I just need to have msys2 for Ruby gtk2/3 support. So, I manually deleted the binaries those are not used by gtk & the supporting gems from msys2 (by checking the dependencies on https://packages.msys2.org/ for gtk ruby). That cuts down the size to about 150MB. But this is not a clean way.

Thus, was wondering if I could be able to generate the gems those include the needed binaries.

Best regards
Tex

**Reply by kojix2** [reply]
Thank you. The problem is a little clearer now.
I don't know the answer to this question, so kou will answer it for you.
Let me ask you one more question: what is the reason for creating a gem?
For example, do you plan to upload it to the rubygem server?

**Reply by kojix2** [reply]
Also, if you want to build small and simple tools, [LibUI](https://github.com/kojix2/LibUI) and the [Glimmer DSL for LibUI](https://github.com/AndyObtiva/glimmer-dsl-libui) may be useful.

**Reply by TexDh** [reply]
Hi Kojix,
No, I don't want to upload it to the rubygem server. I want it solely for my own implementation at this point.
In general, I don't need to build a gem. If I have all the resources and infrastructure that can be used without msys2 then that works for me. 

But I will give it a try with your recommendations.
Thanks for the swift replies.

**Reply by TexDh** [reply]
Or the qtn should have been:
How can I use gtk2/3 without msys2?

**Reply by kojix2** [reply]
> How can I use gtk2/3 without msys2?

I think you have found the correct question.
Kou, the developer of ruby-gnome, is a Linux user and would not normally use Windows. So I don't think he would be able to answer the Windows question straight away either. I don't think anyone here knows much about how to install GTK standalone on Windows, so you'll have to look elsewhere. And once you have found a way to install GTK3 on Windows, the next question is: "I have installed GTK3 on Windows standalone in this way, but can I call it from ruby-gnome?"

**Reply by TexDh** [reply]
Many thanks @kojix2.
Looking back, all the binaries were included as 'x64-mingw32' until gtk2/3 gem version 3.2.9. Those would run without msys2. 
So, I'm looking for the similar approach. Those gems are owned and pushed by @kou. 
Hopefully Kou would be able to guide me in.

**Reply by kou** [reply]
It's created by cross-compile on my local Linux machine.
I don't recommend that you use the approach because it's too difficult to maintain. (Because I dropped the approach.)

**Comment by kou** [comment]
You may be able to extract needed files from installed for GTK 3 by using `pacman -Ql mingw-w64-ucrt-x86_64-gtk3` and its dependencies. You can try copying only the needed files to other environment to work without MSYS2.

**Reply by TexDh** [reply]
Thanks for getting back!
I will give it a try. 

For now I am manually deleting all the not used files by GTK and it's dependencies from msys2.
But that's really cumbersome and I never know whether it gonna sustain.
